pyiceberg/table/snapshots.py

```diff
@@ -116,3 +144,199 @@ class MetadataLogEntry(IcebergBaseModel):
+        elif data_file.content == DataFileContent.POSITION_DELETES:
+            self.removed_delete_files += 1
+            self.removed_pos_delete_files += 1
+            self.removed_pos_deletes += data_file.record_count
+        elif data_file.content == DataFileContent.EQUALITY_DELETES:
+            self.removed_delete_files += 1
+            self.removed_eq_delete_files += 1
+            self.removed_eq_deletes += data_file.record_count
+        else:
+            raise ValueError(f"Unknown data file content: {data_file.content}")
+
+    def added_manifest(self, manifest: ManifestFile) -> None:
+        if manifest.content == ManifestContent.DATA:
+            self.added_files += manifest.added_files_count or 0
+            self.added_records += manifest.added_rows_count or 0
+            self.removed_files += manifest.deleted_files_count or 0
+            self.deleted_records += manifest.deleted_rows_count or 0
+        elif manifest.content == ManifestContent.DELETES:
+            self.added_delete_files += manifest.added_files_count or 0
+            self.removed_delete_files += manifest.deleted_files_count or 0
+        else:
+            raise ValueError(f"Unknown manifest file content: {manifest.content}")
+
+    def build(self) -> Dict[str, str]:
+        def set_non_zero(properties: Dict[str, str], num: int, property_name: str) -> None:
+            if num > 0:
+                properties[property_name] = str(num)
+
+        properties: Dict[str, str] = {}
+        set_non_zero(properties, self.added_size, 'added-files-size')
+        set_non_zero(properties, self.removed_size, 'removed-files-size')
+        set_non_zero(properties, self.added_files, 'added-data-files')
+        set_non_zero(properties, self.removed_files, 'removed-data-files')
+        set_non_zero(properties, self.added_eq_delete_files, 'added-equality-delete-files')
+        set_non_zero(properties, self.removed_eq_delete_files, 'removed-equality-delete-files')
+        set_non_zero(properties, self.added_pos_delete_files, 'added-position-delete-files')
+        set_non_zero(properties, self.removed_pos_delete_files, 'removed-position-delete-files')
+        set_non_zero(properties, self.added_delete_files, 'added-delete-files')
+        set_non_zero(properties, self.removed_delete_files, 'removed-delete-files')
+        set_non_zero(properties, self.added_records, 'added-records')
+        set_non_zero(properties, self.deleted_records, 'deleted-records')
+        set_non_zero(properties, self.added_pos_deletes, 'added-position-deletes')
+        set_non_zero(properties, self.removed_pos_deletes, 'removed-position-deletes')
+        set_non_zero(properties, self.added_eq_deletes, 'added-equality-deletes')
+        set_non_zero(properties, self.removed_eq_deletes, 'removed-equality-deletes')
+
+        return properties
+
+
+properties = ['records', 'files-size', 'data-files', 'delete-files', 'position-deletes', 'equality-deletes']
```

Comment:
Seems this is not used?

Reply:
Good one. I wanted to loop over the properties, but discovered that we both use `delete` and `removed` intertwined